pkg/machines/components/deleteDialog.jsx

```diff
@@ -103,16 +115,19 @@ export class DeleteDialog extends React.Component {
         Object.keys(vm.disks).sort()
                 .forEach(t => {
                     let d = vm.disks[t];
-                    if (d.type == 'file' && d.source.file)
-                        disks.push({ target: d.target, file: d.source.file, checked: !d.readonly });
+                    let entry = d;
+
+                    entry.checked = !d.readonly;
+                    if ((d.type == 'file' && d.source.file) || d.type == 'volume')
+                        disks.push(entry);
                 });
         this.setState({ showModal: true, disks: disks, destroy: vm.state === 'running' });
     }
 
     delete() {
         let storage = [ ];
 
-        this.state.disks.forEach(d => { if (d.checked) storage.push(d.file); });
+        this.state.disks.forEach(d => { if (d.checked) storage.push(d); });
```

Comment:
Stylistic nitpick, but using `const storage = this.state.disks.filter(d => d.checked);` is a bit more straightforward and possibly more efficient.

Reply:
+1 ty